### DISCUSSION TITLE
Meaning of filename and conventions for the name

### DISCUSSION THREAD

**Discussion by nmayle** [CMSgov/price-transparency-guide#50] [Q&A] [answered]
Since a filename is not a concept that has an obvious meaning on a website, what do you mean when you specify the filename?  Do you mean that is the filename should be a component of the URL or that the filename should be the name that the file gets set to when it is downloaded or both?  Is it intended to make the file easier to find on the website?

There are also several other issues with the specifics of the filename and how the issuer and plan name are treated. For example, how are capital letters in the issuer name treated?  Should the filename, as in the examples, use a lowercase version of the issuer and plan name?  How about non-alphabetic characters?

**Comment by shaselton-usds** [comment, marked as answer]
This is going to sound a bit circular, but the filename is meant to be the name of the file that is downloaded. It has nothing to do with a URL/URI. 

At the moment, there are no requirements to require uppercase or lowercase names. 

As for non-alphabetic/numeric characters: would you have an example? There are really only two variables where this might happen, the name of the plan and the name of the company.

**Reply by nmayle** [reply]
Why are you standardizing the filename at all?

There seem to be two different possible benefits from having a standard for the file names and in particular one that includes the payer and plan.  One benefit would be that once the file is downloaded you can clearly determine what payer and plan this corresponds to just by looking at the filename.  Also you can would know which of the three types of files each file is.  All of this information (payer, plan, and disclosure type) could just as easily be in the file in a defined field.
All of the fields other than the disclosure type (in network, allowed, prescription) are already inside of the files in the current design.

The other possible benefit of having a standard filename is to help in finding the files on the website of the payer.  If you want it to be easier to find the files, then you need to tightly specify the name of the url for each file or at least the filename when downloaded.  This would include specifying how to handle capitalization.

An example of a special character in a name of a plan is a single-quote.

**Reply by tdfow** [reply]
would $ or () be allowed or would you need to have the name without those special characters, for example:  2020-01-05_issuer-abc_healthcare-$100_(expanded)_in-network-rates.json

**Reply by shaselton-usds** [reply]
You're right -- including a single quote in a filename is going to be problematic. 

Updates have been included here: https://github.com/CMSgov/price-transparency-guide/commit/6acf1ae1990673d47ce20c01f80a2fae2be32369

Thanks for the thoughts.
